test/repositories/identity.test.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * @jest-environment node
+ */
+
+import { user } from '../factories/user';
+
+import * as userRepository from '@/repositories/user';
+import * as identityRepository from '@/repositories/identity';
+import { AuthProviders } from '@/constants/enum';
+
+describe('identity:', () => {
+  describe('creates identity', () => {
+    describe('given valid parameter', () => {
+      it('creates identity', async () => {
+        const userId = 1;
+        const userPayload = { ...user, id: userId };
+        await userRepository.createUser(userPayload);
+        const identityPayload = {
+          user: { connect: { id: userId } },
+          provider: AuthProviders.GOOGLE,
+        };
+        const createdIdentity = await identityRepository.createIdentity(identityPayload);
+
+        expect(createdIdentity.userId).toEqual(userId);
+        expect(createdIdentity.provider).toEqual(AuthProviders.GOOGLE);
+      });
+    });
+  });
+
+  describe('find identity', () => {
+    describe('given identity is present', () => {
+      describe('given identity relates to user', () => {
+        it('returns the identity', async () => {
+          const userId = 1;
+          const userPayload = { ...user, id: userId };
+          await userRepository.createUser(userPayload);
+          const identityPayload = {
+            user: { connect: { id: userId } },
+            provider: AuthProviders.GOOGLE,
+          };
+          await identityRepository.createIdentity(identityPayload);
+
+          const [identity] = await identityRepository.findByProvider(AuthProviders.GOOGLE, userId);
+
+          expect(identity).not.toBeNull();
+        });
+      });
+
+      describe('given identity does not relates to user', () => {
```

Comment:
English syntax issue + tip to emphasize the sad path:

```suggestion
      describe('given identity does NOT relate to user', () => {
```